src/llama_stack/core/server/tracing.py

```diff
@@ -21,6 +23,25 @@ def __init__(self, app, impls, external_apis: dict[str, ExternalApiSpec]):
         # FastAPI built-in paths that should bypass custom routing
         self.fastapi_paths = ("/docs", "/redoc", "/openapi.json", "/favicon.ico", "/static")
 
+    def _is_router_based_route(self, path: str) -> bool:
+        """Check if a path belongs to a router-based API.
+
+        Router-based APIs use FastAPI routers instead of the old webmethod system.
+        We need to check if the path matches any router-based API prefix.
+        """
+        # Extract API name from path (e.g., /v1/batches -> batches)
+        # Paths are typically /v1/{api_name} or /v1/{api_name}/...
+        parts = path.strip("/").split("/")
+        if len(parts) >= 2 and parts[0].startswith("v"):
```

Comment:
```suggestion
        if len(parts) >= 2 and parts[0].startswith("v1"):
```

nit, but all the routes start with `v1`, `v1alpha`, `v1beta`. Makes this a bit more strict. Could even go with a stronger check to ensure it's one of our valid API levels.

Reply:
but we just need `v` right? if we add another level, we would need to edit this code, I'm wondering how worth it that change is. this is a heuristic to decide routing, not validation. how strong do you feel about this? :)

Reply:
this just reads very fragile -- can you actually test explicitly against `v1`, `v1alpha` and such?